src/VirtualClient/VirtualClient.Dependencies/AMDGPUDriverInstallation.cs

```diff
@@ -0,0 +1,139 @@
+        /// <param name="parameters">A series of key value pairs that dictate runtime execution.</param>
+        public AMDGPUDriverInstallation(IServiceCollection dependencies, IDictionary<string, IConvertible> parameters)
+            : base(dependencies, parameters)
+        {
+            dependencies.ThrowIfNull(nameof(dependencies));
+            this.RetryPolicy = Policy.Handle<Exception>().WaitAndRetryAsync(5, (retries) => TimeSpan.FromSeconds(retries + 1));
+            this.systemManager = dependencies.GetService<ISystemManagement>();
+            this.stateManager = this.systemManager.StateManager;
+            this.fileSystem = this.systemManager.FileSystem;
+            this.packageManager = this.systemManager.PackageManager;
+        }
+
+        /// <summary>
+        /// Retrieves the interface to interacting with the underlying system.
+        /// </summary>
+
+        /// <summary>
+        /// Determines whether Reboot is required or not after Driver installation
+        /// </summary>
+        public bool RebootRequired
+        {
+            get
+            {
+                switch (this.Platform)
+                {
+                    case PlatformID.Win32NT:
+                        return this.Parameters.GetValue<bool>(nameof(AMDGPUDriverInstallation.RebootRequired), false);
+                    default:
+                        return this.Parameters.GetValue<bool>(nameof(AMDGPUDriverInstallation.RebootRequired), true);
+                }
+            }
+        }
+
+        /// <summary>
+        /// A policy that defines how the component will retry when
+        /// it experiences transient issues.
+        /// </summary>
+        public IAsyncPolicy RetryPolicy { get; set; }
+
+        /// <summary>
+        /// Executes  GPU driver installation steps.
+        /// </summary>
+        /// <returns></returns>
+        protected override async Task ExecuteAsync(EventContext telemetryContext, CancellationToken cancellationToken)
+        {
+            this.Logger.LogTraceMessage($"{this.TypeName}.ExecutionStarted", telemetryContext);
+            State installationState = await this.stateManager.GetStateAsync<State>(nameof(AMDGPUDriverInstallation), cancellationToken)
+                .ConfigureAwait(false);
+            if (installationState == null)
+            {
+                if (this.Platform == PlatformID.Win32NT)
+                {
+                    await this.AMDDriverInstallationOnWindowsAsync(telemetryContext, cancellationToken)
+                               .ConfigureAwait(false);
+                    await this.stateManager.SaveStateAsync(nameof(this.AMDDriverInstallationOnWindowsAsync), new State(), cancellationToken)
+                        .ConfigureAwait(false);
+                }
+
+                VirtualClientRuntime.IsRebootRequested = this.RebootRequired;
+            }
+
+            this.Logger.LogTraceMessage($"{this.TypeName}.ExecutionCompleted", telemetryContext);
```

Comment:
Please do check other executors, how you can only have one log and that will print function start and stop and no need of explicitly mentioning